spot_driver_cpp/src/spot_image_publisher.cpp

```diff
@@ -0,0 +1,144 @@
+// Copyright (c) 2023 Boston Dynamics AI Institute LLC. All rights reserved.
+
+#include <spot_driver_cpp/spot_image_publisher.hpp>
+
+#include <rclcpp/node.hpp>
+#include <rmw/qos_profiles.h>
+#include <sensor_msgs/msg/camera_info.hpp>
+#include <sensor_msgs/msg/image.hpp>
+#include <spot_driver_cpp/interfaces/rclcpp_parameter_interface.hpp>
+#include <spot_driver_cpp/interfaces/rclcpp_publisher_interface.hpp>
+#include <spot_driver_cpp/interfaces/rclcpp_tf_interface.hpp>
+#include <spot_driver_cpp/interfaces/rclcpp_wall_timer_interface.hpp>
+#include <spot_driver_cpp/interfaces/spot_interface.hpp>
+#include <spot_driver_cpp/spot_image_sources.hpp>
+#include <spot_driver_cpp/types.hpp>
+
+#include <iostream>
+#include <memory>
+#include <optional>
+#include <stdexcept>
+
+namespace
+{
+constexpr auto kImageCallbackPeriod = std::chrono::duration<double>{1.0 / 15.0 };  // 15 Hz
+constexpr auto kDefaultDepthImageQuality = 100.0;
+}  // namespace
+
+namespace spot_ros2
+{
+::bosdyn::api::GetImageRequest createImageRequest(const std::vector<ImageSource>& sources, [[maybe_unused]] const bool has_rgb_cameras, const double rgb_image_quality, const bool get_raw_rgb_images)
+{
+    ::bosdyn::api::GetImageRequest request_message;
+
+    for (const auto& source : sources)
+    {
+        const auto source_name = toSpotImageSourceName(source);
+        if (source.type == SpotImageType::RGB)
+        {
+            bosdyn::api::ImageRequest* image_request = request_message.add_image_requests();
+            image_request->set_image_source_name(source_name);
+            image_request->set_quality_percent(rgb_image_quality);
+            image_request->set_pixel_format(bosdyn::api::Image_PixelFormat_PIXEL_FORMAT_RGB_U8);
+            image_request->set_image_format(get_raw_rgb_images ? bosdyn::api::Image_Format_FORMAT_RAW : bosdyn::api::Image_Format_FORMAT_JPEG);
+        }
+        else if (source.type == SpotImageType::DEPTH)
+        {
+            bosdyn::api::ImageRequest* image_request = request_message.add_image_requests();
+            image_request->set_image_source_name(source_name);
+            image_request->set_quality_percent(kDefaultDepthImageQuality);
+            image_request->set_image_format(bosdyn::api::Image_Format_FORMAT_RAW);
+        }
+        else // SpotImageType::DEPTH_REGISTERED
```

Comment:
```suggestion
        else  // SpotImageType::DEPTH_REGISTERED
```
Two spaces between code and comments